cats-effect/shared/src/main/scala/cats/effect/std/ArbitraryRandom.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats.effect.std
+
+import cats.effect.*
+import cats.effect.std.Random.ScalaRandom
+import cats.syntax.all.*
+import org.scalacheck.{Arbitrary, Gen, Shrink}
+
+trait ArbitraryRandom {
+
+  /** Generates a `Random[F]` instance seeded by a generated long value, so that the "randomness"
+    * will be repeatable given the same ScalaCheck seed.
+    */
+  def genRandom[F[_]: Sync]: Gen[Random[F]] =
+    Gen.long
+      .map(new scala.util.Random(_).pure[F])
+      .map(new ScalaRandom[F](_) {})
```

Comment:
To be honest, it looks a bit "hacky" to me: `ArbitraryRandom` is placed into `cats.effect.std` to let it access the `ScalaRandom`'s private constructor. Whereas the other `scalacheck-effect` code lives in `org.scalacheck.effect`.

Reply:
Yes, that's true, but I did ask about this implementation in Discord and got an initial thumbs up from @djspiewak. (Although to be fair, I’m not sure he was thinking about it from this specific angle.) I suppose if Cats Effect could be persuaded to open up one of those constructors, we could move this somewhere else.

Really the problem is that the `Random.scalaUtilRandomSeedLong` returns `F[Random[F]]` and not just `Random[F]`. I don’t think that’s necessary for this particular case, but I’m not sure whether that’s true in general. If introducing a new constructor in `Random` is a better option, then we could use that here instead.

Reply:
In regards to constructors safety, there are a couple of classes in Cats-Effect that got "unsafe" constructor counterparts: `Ref` and `Deferred`. Both are in "kernel" though. However, I don't think that it would hurt much if `Random` could get a couple of `unsafe` constructors too. At least to me it seems like a more viable solution comparing to tuning package names in order to access private constructors.

For example, `Random.javaUtilRandom[F[_]](random: java.util.Random): F[Random[F]]` is not fully referentially transparent, apparently. So, it wouldn't seem too bad if there was `unsafeJavaUtilRandom[F[_]](random: java.util.Random): Random[F]` counterpart.